src/Transmitron/Models/Layouts.cpp

```diff
@@ -0,0 +1,462 @@
+  bool isDir = fs::is_directory(mLayoutsDir);
+
+  std::error_code ec;
+  if (exists && !isDir && !fs::remove(mLayoutsDir, ec))
+  {
+    wxLogWarning("Could not remove file %s: %s", mLayoutsDir, ec.message());
+    return false;
+  }
+
+  if (!exists && !fs::create_directory(mLayoutsDir))
+  {
+    wxLogWarning(
+      "Could not create profiles directory: %s",
+      mLayoutsDir
+    );
+    return false;
+  }
+
+  for (const auto &entry : fs::directory_iterator(mLayoutsDir))
+  {
+    wxLogMessage("Checking %s", entry.path().u8string());
+
+    // Get name.
+    std::vector<uint8_t> decoded;
+    try
+    {
+      decoded = cppcodec::base32_rfc4648::decode(
+        entry.path().stem().u8string()
+      );
+    }
+    catch (cppcodec::parse_error &e)
+    {
+      wxLogError(
+        "Could not decode '%s': %s",
+        entry.path().u8string(),
+        e.what()
+      );
+      continue;
+    }
+    const std::string name{decoded.begin(), decoded.end()};
+
+    std::ifstream input(entry.path());
+    if (!input.is_open())
+    {
+      wxLogWarning("Could not open '%s'", entry.path().u8string());
+      continue;
+    }
+
+    std::stringstream buffer;
+    buffer << input.rdbuf();
+    if (!nlohmann::json::accept(buffer.str()))
+    {
+      wxLogWarning("Could not parse '%s'", entry.path().u8string());
+      continue;
+    }
+
+    const auto j = nlohmann::json::parse(buffer.str());
+    const auto perspectiveIt = j.find("perspective");
+    if (
+      perspectiveIt == std::end(j)
+      || perspectiveIt->type() != nlohmann::json::value_t::string
+    ) {
+      wxLogWarning("Could not parse '%s'", entry.path().u8string());
+      continue;
+    }
+
+    const auto id = mAvailableId++;
+    auto layout = std::make_unique<Node>();
+    layout->name        = name;
+    layout->perspective = perspectiveIt->get<std::string>();
+    layout->path        = entry.path();
+    layout->saved       = false;
+    mLayouts.insert({id, std::move(layout)});
+
+    const auto parent = wxDataViewItem(nullptr);
+    const auto item = toItem(id);
+    ItemAdded(parent, item);
+  }
+
+  return true;
+}
+
+wxDataViewItem Layouts::getItem(const std::string &name) const
+{
+  const auto it = std::find_if(
+    std::begin(mLayouts),
+    std::end(mLayouts),
+    [name](const auto &layout){
+      return layout.second->name == name;
+    }
+  );
+
+  if (it == std::end(mLayouts))
+  {
+    return wxDataViewItem(nullptr);
+  }
+
+  return toItem(it->first);
+}
+
+bool Layouts::remove(wxDataViewItem item)
+{
+  const auto id = toId(item);
+  const auto &node = mLayouts.at(id);
+
+  std::error_code ec;
+  fs::remove_all(node->path, ec);
+  if (ec)
+  {
+    wxLogError("Could not delete '%s': %s", node->name, ec.message());
+    return false;
+  }
+
+  mLayouts.erase(id);
+
+  const auto parent = wxDataViewItem(nullptr);
+  ItemDeleted(parent, item);
+
+  return true;
+}
+
+std::optional<std::string> Layouts::getLayout(const std::string &name) const
```

Comment:
It's unclear what the function does just by looking at it's signature.

- Change the argument to wxDataViewItem.
- Remove the std::optional (there is no reason for it to fail using wxDataViewItem).
- Create typedef for the return value (`Perspective`?).